### DISCUSSION TITLE
Highlights in editor

### DISCUSSION THREAD

**Discussion by JarichVanMaldegem** [pwndoc/pwndoc#409] [Ideas]
Hi,

I'm working with pwndoc for quite a while now and I thought it would be nice to have the capability to highlight pieces of text in the editor field. This way it should be albe to highlight a piece of code as evidence. Is this something that can be implemented in the rich text editor? Another plus would be to have some kind of track changes capability for the issues. 

Thanks in advance!

**Comment by Zeecka** [comment]
There is already 2 codes styles implemented on the text editor (Blockquote and CodeBlock). Are you looking for a Highlited CodeBlock ? PwnDoc aim to use [default Word style](https://pwndoc.github.io/pwndoc/#/docxtemplate?id=styles) so the user just have to modify his Office Template to change the expected style (see "Code" for the CodeBlock style).  https://github.com/pwndoc/pwndoc/blob/203055b6af5ab58f3a170dccef23854291f40057/backend/src/lib/html2ooxml.js#L43-L46 I think it'll be possible to implement such feature. From a front-end pov, there is a [CodeBlockHighlight](https://github.com/ueberdosis/tiptap/blob/v1/packages/tiptap-extensions/src/nodes/CodeBlockHighlight.js) plugin for the tiptap WYSIWYG which could be imported in the editor configuration. 
https://github.com/pwndoc/pwndoc/blob/265e359cf2383794ce411193e3526270cbae8902/frontend/src/components/editor.vue#L197-L214
I don't think there will be any breaking change but it might take some times, i'll work on it and make a PR.
